### PR TITLE
[SONIC-1384] Build BpkFlightLeg in Backpack

### PR DESCRIPTION
**Pull request by michmaml**
# JIRA
https://skyscanner.atlassian.net/browse/SONIC-1384

# Why these changes
Flight Leg was the last unbuilt component in Backpack. This PR only contains its implementation in UIKit and SwiftUI's version will be built in the future. This pull request introduces the actual BPK component, a controller which generates four examples in _Example/_ folder and a test file to tests the snapshots.

## Screenshots
**Light Mode** | **Dark Mode** 
---|---
light mode | dark mode

_<sub><sup>This is my first iOS PR so any feedback, comments and corrections are welcome</sup></sub>_
---
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Review by gert-janvercauteren — COMMENTED** [review]
Could you please add a README.md for this new component?

**Comment by michmaml** [comment]
^ Just added the [README](https://github.com/Skyscanner/backpack-ios/blob/sonic/SONIC-1384-bpk-flightleg-uikit/Backpack/FlightLeg/README.md). 😄

File: Backpack/FlightLeg/Classes/BPKFlightLeg.swift [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        NSLayoutConstraint.activate([
+            leadingStackView.topAnchor.constraint(equalTo: verticalStackView.topAnchor, constant: BPKSpacingSm),
+            leadingStackView.bottomAnchor.constraint(equalTo: verticalStackView.bottomAnchor),
+            leadingStackView.leadingAnchor.constraint(equalTo: verticalStackView.leadingAnchor),
+            leadingStackView.trailingAnchor.constraint(equalTo: verticalStackView.trailingAnchor)
+        ])
+        return verticalStackView
+    }
+    
+    private func setupFlightStackView() -> UIView {
```
Comment (frugoman) [inline Backpack/FlightLeg/Classes/BPKFlightLeg.swift:187]: can you rename this function to reflect that's actually creating flight stack? something like `createFlightStackView` or `makeFlig...`

File: Backpack/FlightLeg/Classes/BPKFlightLeg.swift [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            .font: BPKFont.makeFont(fontStyle: .textCaption),
+            .foregroundColor: BPKColor.textOnLightColor
+        ]
+    }
+    
+    public static func descriptionDefaultAttributes() -> [NSAttributedString.Key: Any] {
+        return [.font: BPKFont.makeFont(fontStyle: .textCaption)]
+    }
+    
+    private func setupDurationStackView() -> UIView {
```
Comment (frugoman) [inline Backpack/FlightLeg/Classes/BPKFlightLeg.swift:232]: same as above, if it's setting up, there's no need to return it.

I would keep the setup as is maybe and just have it not return anything.

**Review by frugoman — CHANGES_REQUESTED** [review]
BPKFlightLegSnapshotTest.swift is in the wrong folder and this made the snapshots be generated in the wrong place. Please relocate the files!

File: Backpack/FlightLeg/Classes/BPKFlightLeg.swift [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+public final class BPKFlightLeg: UIView {
+    private var departureArrivalTime: String
+    private var nextDayArrival: String?
+    private var flightDescription: NSAttributedString
+    private var stopsInfo: String
+    private var highlightStopsInfo: Bool = false
+    private var duration: String
+    private var operatedBy: String?
+    private var warning: String?
+    private var carrierLogo: BPKIconName
```
[the remark this answers is not in the document]
Comment (michmaml) [reply Backpack/FlightLeg/Classes/BPKFlightLeg.swift:31]: Thank you for pointing it out! I am not sure if `highlightStopsInfo` can be set to a constant, as the line 140 shows an error _Immutable value 'self.highlightStopsInfo' may only be initialized once_.

**Review by frugoman — APPROVED** [review]

**Pull request merged**